misk-hibernate/src/test/kotlin/misk/hibernate/AggregationQueryTest.kt

```diff
@@ -223,6 +224,60 @@ class AggregationQueryTest {
     )
   }
 
+  @Test fun `aggregations work when there are no matching rows`() {
+    primitiveTransacter.transaction {session ->
+      queryFactory.newQuery<PrimitiveTourQuery>()
+        .delete(session)
+    }
+    primitiveTransacter.transaction { session ->
+      // If the query returns no rows, the result should be null.
+      queryFactory.newQuery<PrimitiveTourQuery>()
+        .averageI64(session)
+        .let { assertThat(it).isNull() }
+
+      // If the query returns all null, and the projection cannot accept all nulls, it should
+      // return an empty list.
+      queryFactory.newQuery<PrimitiveTourQuery>()
+        .listI1C16AndMaxI8(session)
+        .let { assertThat(it).isEmpty()}
+
+      // If the query returns all nulls and the projection can accept all nulls, it should return
+      // an instance of the projection.
+      queryFactory.newQuery<NoDataQuery>()
+        .listAllNullsAreOk(session)
+        .let { assertThat(it).containsExactly(AllNullsAreOk(null, null)) }
+
+      queryFactory.newQuery<NoDataQuery>()
+        .allNullsAreOk(session)
+        .let { assertThat(it).isEqualTo(AllNullsAreOk(null, null)) }
+    }
+  }
+
+  @Test fun `query throws when result cannot be assigned to projection`() {
+    primitiveTransacter.transaction {session ->
+      queryFactory.newQuery<PrimitiveTourQuery>()
+        .delete(session)
+    }
+    primitiveTransacter.transaction { session ->
+      assertThat(assertThrows<IllegalArgumentException> {
+        queryFactory.newQuery<NoDataQuery>()
+          .wrongNullability(session)
+      }).hasMessage(
+        """
+        |There were problems with your Projection class:
+        |  Constructor parameter maybeI1 is not nullable, but query returned null for maybeI1. Mark this parameter as nullable to avoid this error.""".trimMargin()
+      )
+      assertThat(assertThrows<IllegalArgumentException> {
+        queryFactory.newQuery<NoDataQuery>()
+          .listWrongNullability(session)
+      }).hasMessage(
+        """
+        |There were problems with your Projection class:
+        |  Constructor parameter maybeI1 is not nullable, but query returned null for maybeI1. Mark this parameter as nullable to avoid this error.""".trimMargin()
+      )
+    }
+  }
```

Comment:
TODO: This test actually belongs with ReflectionQuery (it has nothing to do with aggregations!) and should operate on the movie transacter. But it works!

Reply:
Hmm this test might not actually be that useful. It worked locally, but then I discovered my mysql was running in a weird sql_mode=NO_ENGINE_SUBSTITUTION, instead of the default, which forbids the types of queries which can result in these problems.

I think I'll remove it, along with the checks that attempt to match up nullability in projection properties.